### PR TITLE
Remove Qt Deprecated Warnings Part 3

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Description

QXmlDefaultHandler -> QXmlStreamReader

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added my user impacting change to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document. 

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/control/objs/BundleSettings/unitTest.cpp [hunk 1]
```diff
@@ -83,7 +82,6 @@ namespace Isis {
       /**
        * Constructs BundleSettings using XML handler.
        *
-       * @param project A pointer to the project.
        * @param reader A pointer to a XmlStackedHandlerReader.
```
Comment (chkim-usgs) [inline isis/src/control/objs/BundleSettings/unitTest.cpp]: We could change `XmlStackedHandlerReader` to `QXmlStreamReader`.

**Review by chkim-usgs — COMMENTED** [review]
Looks good otherwise.

**Review by chkim-usgs — DISMISSED** [review]

**Comment by amystamile-usgs** [comment]
The following tests failing:

- deltack_app_test_hist (Failed)
- deltack_app_test_singlePoint (Failed)
- deltack_app_test_twist (Failed)
- ThreeImageNetwork.BundleSolutionInfoSerialization (SEGFAULT)
- DefaultCube.FunctionalTestCaminfoCsv (Timeout)
- ObservationPair.FunctionalTestJigsawCamSolveAll (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawApollo (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawBundleXYZ (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawHeldList (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawOutlierRejection (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawMEstimator (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawPoleRaDecW0WdotMeanRadius (Subprocess aborted)
- ApolloNetwork.FunctionalTestJigsawPoleRaDecW0WdotTriaxial (Subprocess aborted)
- VikThmNetwork.FunctionalTestJigsawScconfig (Subprocess aborted)
- VikThmNetwork.FunctionalTestJigsawScconfigHeld (Subprocess aborted)
- MiniRFNetwork.FunctionalTestJigsawRadar (Subprocess aborted)
- CSMNetwork.FunctionalTestJigsawCSM (Subprocess aborted)
- LidarNetwork.FunctionalTestJigsawLidar (Subprocess aborted)

**Comment by amystamile-usgs** [comment]
The follow test is failing:

`1201 - ThreeImageNetwork.BundleSolutionInfoSerialization (SEGFAULT)`

**Comment by amystamile-usgs** [comment]
Tests are passing.

**Review by chkim-usgs — APPROVED** [review]
Tested again on my local Mac machine, looks good.

**Pull request merged**